### DISCUSSION TITLE
Not a bad start for a beta version, but alot of issues.

### DISCUSSION THREAD

**Discussion by cranyy** [plandex-ai/plandex#44] [General]
Its incredibly ineffective at actually creating proper large projects. It fills tens of files with broken stuff or indent errors, and then you have to painstakingly write "plandex tell" for each stupid error again and again. Why doesnt it test its own shit at any point?

 It creates basically files for every single function, which makes it supremely hard and annoying to manually fix the errors, and it always, always fails at setting up paths correctly for Windows it often writes them like this "EProjectTestplan" 

I got this error that proved impossible for it to fix after 5 tries now - ValueError: The filepath provided must end in `.keras` (Keras model format). Received: filepath=E:\Project\testplan\model_weights.h5" literally the only thing it had to do was change .h5 to .keras to fix it and it couldnt figure that out 5 times in a row with clear explanations on my end. 

Not a bad beta, but definitely very far away from a self-sustained service.

**Comment by danenania** [comment]
Thanks for trying it and the feedback @cranyy. I moved this over to a discussion since it's more general feedback than a specific issue.

One suggestion is not to try to use `plandex tell` to fix mistakes in file updates. Check out the [limitations and guidance](https://github.com/plandex-ai/plandex?tab=readme-ov-file#limitationsand-guidance-%EF%B8%8F) section in the readme. For now it's usually more efficient to reject the file in the `plandex changes` TUI and apply that change yourself.

It also doesn't yet have a concept of renaming a file, so it makes sense that it could have difficulty on changing a file extension as you describe.

Screwing up windows paths sounds like something that could potentially be fixed with some prompting and examples. I'll investigate that. Did you do this testing on cloud or self-hosted? If cloud, I'll look for your plans to use as test cases--feel free to send me an email dane@plandex.ai and let me know what email you signed up with if you created an account with your email.

Testing its own shit as you put it is definitely on the roadmap.

**Comment by foragerr** [comment]
@danenania have you considered benchmarking Plandex with [SWE-bench](https://www.swebench.com/) or similar, to get a sense of Plandex's capability against other similar tooling?

**Comment by danenania** [comment]
@cranyy I'd love hear if 0.9.0 (https://github.com/plandex-ai/plandex/releases/tag/cli%2Fv0.9.0) has resolved any of these issues for you. It includes some major improvements to reliability.
